### ISSUE TITLE
Allow for environment specific configuration

### ISSUE THREAD

**Issue by markunsworth** [conda/conda#1957]
It would be useful to allow for an environment specific configuration so that we can append configuration to the user/default condarc based configuration.

We run our CI using conventions so we know a project expects conda if there's a known file (conda.txt) present and we use that file to install the required packages for that application:

```
conda create -p env --file conda.txt
```

however there's often times when someone needs a package from a custom channel. We were updating all our .condarc files to include that channel but have just implemented a solution that looks for a specific file in our application that we can read in as additional arguments during installation:

```
CONDA_ARGS=""
if [ -f conda.args ]; then
    CONDA_ARGS=`cat conda.args`
fi
conda create -p env --file conda.txt $CONDA_ARGS
```

the `conda.args` file is just as you would expect, a set of args to append to the command 

```
-c foo -c bar -c baz
```

However i think it would be really useful to be able to append an additional configuration (in the condarc format) for an individual environment or command on the command line: 

```
conda create -p env --file conda.txt --config conda.config

conda config --env conda.config
```

**Comment by jakirkham** [comment]
Like the general idea, but I feel like this may be a dup of an existing issue.

**Comment by jakirkham** [comment]
Sounds sort of similar to this issue ( https://github.com/conda/conda/issues/1329 ). Though maybe you can clarify whether that sounds accurate or not.

Also, this issue ( https://github.com/conda/conda/issues/2474 ) seems to grouping this stuff together to determine some kind of actionable so discussion is probably better there. @kalefranz, do you have thoughts on this point?

**Comment by kalefranz** [comment]
Implemented in conda 4.2.  See https://www.continuum.io/blog/developer-blog/conda-configuration-engine-power-users
